src/test/java/uk/gov/justice/hmpps/prison/service/OffenderTransactionHistoryServiceTest.java

```diff
@@ -41,301 +40,398 @@ public void setUp() {
+                    .transactionId(1L)
+                    .build(),
+                transaction.toBuilder()
+                    .createDatetime(LocalDateTime.now().minusDays(1))
+                    .entryDate(LocalDate.now().minusDays(1))
+                    .entryAmount(BigDecimal.ONE)
+                    .transactionEntrySequence(1L)
+                    .transactionId(6L)
+                    .build(),
+                transaction.toBuilder()
+                    .createDatetime(LocalDateTime.now().minusDays(1))
+                    .entryDate(LocalDate.now().minusDays(1))
+                    .entryAmount(BigDecimal.ONE)
+                    .transactionEntrySequence(2L)
+                    .transactionId(5L)
+                    .build(),
+                transaction.toBuilder()
+                    .createDatetime(LocalDateTime.now().minusDays(1))
+                    .entryDate(LocalDate.now().minusDays(1))
+                    .entryAmount(BigDecimal.ONE)
+                    .transactionEntrySequence(3L)
+                    .transactionId(4L)
+                    .build(),
+                transaction.toBuilder()
+                    .createDatetime(LocalDateTime.now().minusDays(2))
+                    .entryDate(LocalDate.now().minusDays(2))
+                    .entryAmount(BigDecimal.ONE)
+                    .transactionEntrySequence(1L)
+                    .transactionId(9L)
+                    .build(),
+                transaction.toBuilder()
+                    .createDatetime(LocalDateTime.now().minusDays(2))
+                    .entryDate(LocalDate.now().minusDays(2))
+                    .entryAmount(BigDecimal.ONE)
+                    .transactionEntrySequence(2L)
+                    .transactionId(8L)
+                    .build(),
+                transaction.toBuilder()
+                    .createDatetime(LocalDateTime.now().minusDays(2))
+                    .entryDate(LocalDate.now().minusDays(2))
+                    .entryAmount(BigDecimal.ONE)
+                    .transactionEntrySequence(3L)
+                    .transactionId(7L)
+                    .build()
+            ));
 
-        when(repository.getTransactionHistory(anyString(), any(), any(), any(), any())).thenReturn(Collections.emptyList());
+            final var histories =
+                service.getTransactionHistory(OFFENDER_NO, null, null, null, null);
 
-        final var histories = service.getTransactionHistory(OFFENDER_NO, accountCode, fromDate, null, transactionType);
+            assertThat(histories).isNotNull();
+            assertThat(histories.size()).isEqualTo(9);
 
-        verify(repository, times(1)).getTransactionHistory(OFFENDER_NO, "SPND", fromDate, null, transactionType);
 
-        assertThat(histories).isNotNull();
-        assertThat(histories.size()).isEqualTo(0);
-    }
+            // Date: now, Seq: 3, 2, 1
+            assertThat(histories.get(0).getTransactionId()).isEqualTo(1L);
```

Comment:
Maybe some `extracting` and `containsExactly` might make this a little less verbose?